src/arith_uint256.cpp

```diff
@@ -10,6 +10,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include "crypto/common.h"
```

Comment:
Include ordering is there for a very good purpose, which is to ensure stand-alone headers. This purpose is just not really all that relevant in a project whose headers are not installed and exported for 3rd party use.  Like a DLL based project would need it.

tl;dr dropping the requirement for header-inclusion-ordering makes sense to me.